python/tank/path_cache.py

```diff
@@ -344,15 +344,22 @@ def synchronize(self, log=None, full_sync=False):
             # it could break for example if someone has culled the event log table and in 
             # that case we should fall back on a full sync.
             
-            self._log_debug(log, "Fetching folder event log entries...")
+            self._log_debug(log, "Fetching create/delete folder event log "
+                            "entries >= id %s for project %s..." % (event_log_id, self._get_project_link()))
             
+            # note that we return the records in ascending order, meaning that they get 
+            # "played back" in the same order as they were created.
+            #
+            # for a non-truncated event log table, the first record returned
+            # by this query should be the last one previously processed by the 
+            # path cache (via the event_log_id variable)
             response = self._tk.shotgun.find("EventLogEntry", 
                                              [ ["event_type", "in", ["Toolkit_Folders_Create", 
                                                                      "Toolkit_Folders_Delete"]], 
                                                ["id", "greater_than", (event_log_id - 1)],
                                                ["project", "is", self._get_project_link()] ],
                                              ["id", "meta", "event_type"],
-                                             [{"field_name": "id", "direction": "desc"}] )   
+                                             [{"field_name": "id", "direction": "asc"}] )   
```

Comment:
This is the big, critical change that will cause the sync to be incremental rather than full!